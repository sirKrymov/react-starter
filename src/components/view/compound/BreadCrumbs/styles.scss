.bread-crumbs {
  margin-bottom: 16px;
  align-items: center;
  flex-wrap: wrap;
  display: flex;

  li {
    align-items: center;
    display: flex;
  }

  &__arrow {
    display: block;
    margin: 0 6px;
  }
}
