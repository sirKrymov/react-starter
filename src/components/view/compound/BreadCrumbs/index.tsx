// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { Fragment, ReactElement } from 'react';

import { Typography } from 'components/view/atoms/Typography';
import { SvgIcon } from 'components/view/atoms/SvgIcon';
import { Link } from 'components/view/atoms/Link';

import { visibilityOnIcon } from 'constants/icons';

import './styles.scss';

interface IProps {
  listId: string;
  list: Array<{
    label: string;
    to: string;
  }>;
}

export function BreadCrumbs({ listId, list }: IProps): ReactElement {
  return (
    <ul className="bread-crumbs">
      {list.map((item, index) => {
        if (index === list.length - 1) {
          return (
            <li key={index}>
              <Typography variant="body3">{item.label}</Typography>
            </li>
          );
        }

        return (
          <Fragment key={item.label}>
            <li key={index}>
              <Typography color="textPrimary" variant="body3">
                <Link to={item.to} id={`breadcrumb-${listId}-link-${index}`}>
                  {item.label}
                </Link>
              </Typography>

              <span className="bread-crumbs__arrow">
                <SvgIcon
                  src={visibilityOnIcon}
                  size="xs"
                  rotate="270"
                  color="on-primary"
                />
              </span>
            </li>
          </Fragment>
        );
      })}
    </ul>
  );
}
