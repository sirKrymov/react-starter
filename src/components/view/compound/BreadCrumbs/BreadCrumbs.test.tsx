import React from 'react';

import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import { BreadCrumbs } from '../BreadCrumbs';

describe('<CardWrapper> compound component', () => {
  let component: ReactTestRenderer;

  const list = [
    {
      label: 'Test one',
      to: '/to-test-one'
    },
    {
      label: 'Test two',
      to: '/to-test-two'
    }
  ];

  const createComponent = () => {
    component = renderer.create(
      <BrowserRouter>
        <BreadCrumbs list={list} listId="test-id" />
      </BrowserRouter>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(component.toJSON()).toBeDefined();
    });

    it('THEN it should match snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
