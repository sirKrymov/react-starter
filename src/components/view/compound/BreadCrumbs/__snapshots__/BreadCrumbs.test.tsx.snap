// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardWrapper> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<ul
  className="bread-crumbs"
>
  <li>
    <p
      className="typography typography--display-initial typography--variant-body3 typography--color-textPrimary typography--align-left typography--font-roboto"
    >
      <a
        className="custom-link"
        href="/to-test-one"
        id="breadcrumb-test-id-link-0"
        onClick={[Function]}
      >
        Test one
      </a>
    </p>
    <span
      className="bread-crumbs__arrow"
    >
      <div
        className="svg-icon svg-icon--rotate-270 svg-icon--color-on-primary svg-icon--size-xs"
      />
    </span>
  </li>
  <li>
    <p
      className="typography typography--display-initial typography--variant-body3 typography--color-textPrimary typography--align-left typography--font-roboto"
    >
      Test two
    </p>
  </li>
</ul>
`;
